core/templates/dev/head/pages/exploration-editor-page/editor-tab/exploration-editor-tab.directive.ts

```diff
@@ -92,6 +92,16 @@ angular.module('oppia').directive('explorationEditorTab', [
             }
           }, true);
 
+          $scope.$watch(function() {
+            return StateEditorService.isStateEditorInitialised();
+          }, function() {
+            if (StateEditorService.isStateEditorInitialised() &&
```

Comment:
I think we do similar check inside RouterServices while user switches tab. Why this wasn't required earlier?

Reply:
ExplorationStatesService.isInitialized() is to ensure that the getState method works, it would error if the _states property in ExplorationStatesService is null. StateEditorService.isStateEditorInitialised() is a new method which basically checks all the "*state*EditorInitialised" booleans return true.